core/chains/evm/txmgr/txmgr.go

```diff
@@ -53,13 +53,13 @@ type Config interface {
 }
 
 // KeyStore encompasses the subset of keystore used by txmgr
-type KeyStore interface {
-	CheckEnabled(address common.Address, chainID *big.Int) error
-	EnabledKeysForChain(chainID *big.Int) (keys []ethkey.KeyV2, err error)
-	GetNextNonce(address common.Address, chainID *big.Int, qopts ...pg.QOpt) (int64, error)
-	GetStatesForChain(chainID *big.Int) ([]ethkey.State, error)
-	IncrementNextNonce(address common.Address, chainID *big.Int, currentNonce int64, qopts ...pg.QOpt) error
-	SignTx(fromAddress common.Address, tx *gethTypes.Transaction, chainID *big.Int) (*gethTypes.Transaction, error)
+type KeyStore[ADDR any, ID any, TX any, META any] interface {
+	CheckEnabled(address ADDR, chainID ID) error
+	EnabledKeysForChain(chainID ID) (keys []ethkey.KeyV2, err error) //TODO:remove it and follow GetStatesForChain
+	GetNextMetadata(address ADDR, chainID ID, qopts ...pg.QOpt) (META, error)
```

Comment:
How do you plan to remove qopts from the interface? What I did was I changed I changes it to any any type and type casted it back to pg.QOpt in the implementation. (ex: https://github.com/smartcontractkit/chainlink/pull/8669/files#diff-aab8d00070a71d79a9e196c64f4fed568fe311e2f485cf141007872ca8fbaae6R500) . 

I was also thinking that one could make it a generic parameter, which enforces type safety at compile time.

Reply:
Hmmm what does casting back and forth achieve?
I would like to better understand what `QOpt` is actually doing, and whether we could use standard `context.Context` params instead.

Reply:
I have to check if we can remove QOpt completely from here and wrap them in the ORM directly - if that doesn't work, ill look at the context param - I agree I don't like the idea of using Generics for the QOpt, and its always one type, why generalize that single type?

Reply:
I want to completely remove this function GetNextNonce() from KeyStore.
It don't make sense. The next correct nonce is derived from 2 sources:
- What in the nonce onchain, or in mem-pool
- What is nonce in our txmgr eth_tx_attempts

If these 2 are the source of truth, then why do we need to ask the KeyStore what the nonce is?
We should instead be telling the KeyStore, to sign a TxAttempt with this provided nonce.

The source of truth for next correct Nonce, should always be the TxMgr.

Once we do that, we can remove these functions from the KeyStore entirely:
- GetNextNonce()
- IncrementNextNonce
Also will allow us to get rid of QOpts from this interface, as well as from TxMgr ORM, and unlink these 2 ORMs too.

But let us work on that separately, with @jmank88 .

I would say, till we do that, don't touch these 2 functions, and let them as it is.

Reply:
+1 to leaving this to another PR - i will be taking a look at separating some of these interfaces in my PR for the attempt builder